cpp/src/arrow/flight/sql/odbc/flight_sql/flight_sql_ssl_config.h

```diff
@@ -0,0 +1,62 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <arrow/flight/types.h>
+#include <arrow/status.h>
+#include <string>
+
+namespace driver {
+namespace flight_sql {
+
+/// \brief An Auxiliary class that holds all the information to perform
+///        a SSL connection.
+class FlightSqlSslConfig {
+ public:
+  FlightSqlSslConfig(bool disableCertificateVerification, const std::string& trustedCerts,
+                     bool systemTrustStore, bool useEncryption);
+
+  /// \brief  Tells if ssl is enabled. By default it will be true.
+  /// \return Whether ssl is enabled.
+  bool useEncryption() const;
```

Comment:
Should be either UseEncryption or use_encryption (because this is a trivial getter), ditto for below